hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/TopologySubcommand.java

```diff
@@ -139,6 +151,11 @@ private String getAdditionNodeOutput(HddsProtos.Node node) {
   // Format "ipAddress(hostName):PortName1=PortValue1    OperationalState
   //     networkLocation
   private void printNodesWithLocation(Collection<HddsProtos.Node> nodes) {
+    if (nodeOperationalState != null) {
+      nodes = nodes.stream().filter(
+          info -> info.getNodeOperationalStates(0).toString()
+              .equals(nodeOperationalState)).collect(Collectors.toList());
+    }
```

Comment:
Nit: can you please avoid code duplication by extracting the code block to a method.  It would also make sense to invoke it once from `execute()`, not in each `print...()`.